contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -128,24 +140,349 @@ contract OETHVaultCore is VaultCore {
+        // That is, the amount of WETH that is currently allocated for the withdrawal queue
+        uint256 allocatedWeth = queue.claimable - queue.claimed;
+
+        // If there is no unallocated WETH then there is nothing to add to the queue
+        if (wethBalance <= allocatedWeth) {
+            return 0;
+        }
+
+        uint256 unallocatedWeth = wethBalance - allocatedWeth;
+
+        // the new claimable amount is the smaller of the queue shortfall or unallocated weth
+        addedClaimable = queueShortfall < unallocatedWeth
+            ? queueShortfall
+            : unallocatedWeth;
+        uint256 newClaimable = queue.claimable + addedClaimable;
+
+        // Store the new claimable amount back to storage
+        withdrawalQueueMetadata.claimable = SafeCast.toUint128(newClaimable);
+
+        // emit a WithdrawalClaimable event
+        emit WithdrawalClaimable(newClaimable, addedClaimable);
+    }
+
+    /***************************************
+                View Functions
+    ****************************************/
+
+    /// @dev Calculate how much WETH in the vault is not reserved for the withdrawal queue.
+    // That is, it is available to be redeemed or deposited into a strategy.
+    function _wethAvailable() internal view returns (uint256 wethAvailable) {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // The amount of WETH that is still to be claimed in the withdrawal queue
+        uint256 outstandingWithdrawals = queue.queued - queue.claimed;
+
+        // The amount of sitting in WETH in the vault
+        uint256 wethBalance = IERC20(weth).balanceOf(address(this));
+
+        // If there is not enough WETH in the vault to cover the outstanding withdrawals
+        if (wethBalance <= outstandingWithdrawals) {
+            return 0;
+        }
+
+        return wethBalance - outstandingWithdrawals;
+    }
+
+    /// @dev Get the balance of an asset held in Vault and all strategies
+    /// less any WETH that is reserved for the withdrawal queue.
+    /// This will only return a non-zero balance for WETH.
+    /// All other assets will return 0 even if there is some dust amounts left in the Vault.
+    /// For example, there is 1 wei left of stETH in the OETH Vault but will return 0 in this function.
+    ///
+    /// If there is not enough WETH in the vault and all strategies to cover all outstanding
+    /// withdrawal requests then return a WETH balance of 0
+    function _checkBalance(address _asset)
+        internal
+        view
+        override
+        returns (uint256 balance)
+    {
+        balance = super._checkBalance(_asset);
+
+        if (_asset == weth) {
+            WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+            // Need to remove WETH that is reserved for the withdrawal queue
+            if (balance + queue.claimed >= queue.queued) {
+                return balance + queue.claimed - queue.queued;
```

Comment:
nit: `return` inside conditional statements, Isn't this something that we just agreed not to do to make sure that the code is more readable?

Reply:
you are correct. From what we discussed at the last contract sync, this would be rewritten to

```Solidity
function _checkBalance(address _asset)
    internal
    view
    override
    returns (uint256 balance)
{
    balance = super._checkBalance(_asset);

    if (_asset != weth) {
        return 0;
    }

    WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
    // Need to remove WETH that is reserved for the withdrawal queue
    if (balance + queue.claimed < queue.queued) {
        return 0;
    }

    return balance + queue.claimed - queue.queued;
}
```

But I'm not going to change it at this late stage

Reply:
@DanielVF @shahthepro 
In hind sight I should have done this late change as `_checkBalance` has the same bug as what `_wethAvailable` had. I'm going to change to the above

Reply:
Here's the PR to fix this issue on `master` https://github.com/OriginProtocol/origin-dollar/pull/2162